### PR TITLE
Add build options to UI api; add example to nf_ui.py

### PR DESCRIPTION
**Pull request by MichaelPesce**
## Fixes/Resolves:

- Add build options to UI api. This will allow for configuring build options on the UI in flowsheet export files
- Add example with nf_ui.py
   - add option for with or without bypass
   - add option for concentration polarization type
- Update documentation

## Summary/Motivation:

This will allow developers to build flowsheets in multiple ways with one flowsheet on the UI

## Changes proposed in this PR:
- Update fsapi.py with required changes for build options
- Update nf_ui.py to include new options
- Update all flowsheet export files to include build_options as a parameter in required functions
- Update documentation with updated process for exporting flowsheets to the UI
- Add init file to BSM2_ui directory
- Update fsapi tests

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by MichaelPesce** [comment]
This PR only truly includes **3 file changes** (plus tests): **fsapi**, **nf_ui**, and **how_to_use_ui** doc file. 

All the additional changed files are flowsheet export files that had to be updated to include a new parameter in the build and export functions.

**Review by dangunter — CHANGES_REQUESTED** [review]
add `(..., build_options=None, **kwargs)` in `*_ui.py`

**Review by dangunter — APPROVED** [review]
looks good!

File: watertap/examples/flowsheets/nf_dspmde/nf_ui.py [hunk 1]
```diff
@@ -21,11 +23,27 @@ def export_to_ui():
...
+                "values_allowed": ["false", "true"],
+                "value": "false",
+            },
+            "ConcentrationPolarization": {
+                "name": "ConcentrationPolarization",
+                "display_name": "Concentration Polarization Type",
+                "values_allowed": ["calculated", "none"],
+                "value": "calculated",
+            },
+        },
```
Comment (adam-a-a) [inline watertap/examples/flowsheets/nf_dspmde/nf_ui.py:42]: We can revisit this in a subsequent PR. I think we mainly want to get the functionality in at this point and refine later (e.g., add other relevant options, change the category as needed, etc.)

**Review by adam-a-a — APPROVED** [review]
LGTM - I think there is some refinement that we could make, but that can wait until later.

**Comment by MichaelPesce** [comment]
#1204 - this PR addresses the BSM2_ui flowsheet export by adding the `__init.py__` file to that directory, but does not solve the underlying issue

**Pull request merged**